community/consistency-check/src/main/java/org/neo4j/consistency/ConsistencyCheckService.java

```diff
@@ -243,7 +242,7 @@ public Result runFullConsistencyCheck( final File storeDir, Config config, Progr
 
             LabelScanStore labelScanStore =
                     new NativeLabelScanStore( pageCache, storeDir, FullStoreChangeStream.EMPTY, true, monitors,
-                            RecoveryCleanupWorkCollector.IMMEDIATE );
+                            RECOVERY_PREVENTING_COLLECTOR );
```

Comment:
First of all, it is clear that we should never execute the cleanup job because consistency check should be a read only thing. This change I like!
However, I don't think it is the right thing to block consistency checker from doing it's job since we know that we are not doing a "real" recovery of the store. At the same time, I don't think silently ignoring is a good thing either. Perhaps we could report a dirty tree as any other inconsistency? What do you think?
If this turns out to be a big change, it can wait for some future PR.